### ISSUE TITLE
Problems saving tmap file: Viewport has zero dimension(s)

### ISSUE THREAD

**Issue by pat-s** [mtennekes/tmap#127]
Hi there, 

attended your talk at useR! in Brussels. Thanks for the awesome package! Finally reproducible maps without using a GUI GIS :)

I have trouble saving this tmap file [(link)](https://www.dropbox.com/s/j01jghfx5k9gy86/test.rda?dl=0): 

```r
save_tmap(tm, "some/path/file.pdf", asp = 0, verbose = T)

Error in grid.Call(L_convert, x, as.integer(whatfrom), as.integer(whatto), : Viewport has zero dimension(s)
```

CRAN version 1.10

Any ideas whats wrong?

**Comment by mtennekes** [comment]
Thanks for your question. It turns out to be a hard one:

The first problem is that the pdf device (at least, the one in Ubuntu) doesn't recognize the font family "Fira Sans"

    In addition: Warning messages:
    1: In grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y,  :
    						 	font family 'Fira Sans' not found in PostScript font database

The second problem is that one of the viewports for the external grid labels has zero height. Probably, this has something to do with the aspect ratio.

This one works:

    tm[[14]]$fontfamily <- ""
    tm$tm_grid$grid.labels.inside.frame <- TRUE
    save_tmap(tm, "file.pdf", asp = 0, verbose = T)

Could you post the code that you used to create tm?

**Comment by pat-s** [comment]
Thanks for helping here!

```r
tm_shape(provinces) +
  tm_polygons(col = "grey") +
  tm_text("name", size = "AREA") +
tm_shape(study_area) +
  tm_borders(col = "black", lwd = 3) +
tm_shape(points) +
  tm_symbols(size = 0.2, col = "diplo01", palette = viridis(2, option="D"),
             labels = c("Not Infected", "Infected"),
             title.col = "P. Radiata trees") +
tm_grid(alpha = 0.4, labels.inside.frame = FALSE, col = "white",
        projection = "+proj=longlat +datum=WGS84 +no_defs +ellps=WGS84 +towgs84=0,0,0",
        n.x = 3, labels.col = "black", labels.size = 0.8) + 
tm_credits("Projection: EPSG 4326 (WGS 84)", position = c(0.5, 0.935), 
           col = "white", fontface = "bold") +
tm_compass(type = "4star", position = c("RIGHT","BOTTOM")) + 
tm_scale_bar(size = 0.7, position = c("RIGHT","BOTTOM"), breaks = c(0, 25, 50)) + 
tm_style_natural(bg.color = "dodgerblue3") + 
tm_format_Europe(#main.title = "Spatial distribution of Diplodia Pinea\n infections on Pinus Radiata trees",
                 #main.title.position = c("center", "TOP"),
                 outer.margins = c(0.05,0,0.02,0),
                 main.title.size = 0.9,
                 fontfamily = "Fira Sans") + 
tm_legend(text.size = 0.8,
            title.size = 1,
            position = c("left","bottom"), 
            bg.color = "white", 
            # bg.alpha=.2, 
            frame = "black", 
            height = .6, 
            hist.width = .2,
            hist.height = .2, 
            hist.bg.color = "gray60", 
            hist.bg.alpha = .5) -> tm
```

**Comment by pat-s** [comment]
As a workaround I currently export the plot using the export function in RStudio and convert the resulting .png to .pdf afterwards using `imagemagick`. 

A working `tmap` implementation would still be favoured of course :)

**Comment by mtennekes** [comment]
I found and fixed the error. When plotting a map with outside grid labels, the outer.margins should be positive. However, the default value for `outer.margins` in `save_tmap` is 0, since in most cases the frame should define the edges of the image. I added an error message to explain this. So, outer.margins should be set in `save_tmap`, e.g. with `NA`.

The issue with the unknown font for the pdf device in Ubuntu still applies. I'm not sure if this font is recognized in other OSs.
